org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/ClientPreferences.java

```diff
@@ -189,6 +189,10 @@ public boolean isProgressReportSupported() {
 		return Boolean.parseBoolean(extendedClientCapabilities.getOrDefault("progressReportProvider", "false").toString());
 	}
 
+	public boolean isWorkDoneProgressSupported() {
+		return Boolean.parseBoolean(extendedClientCapabilities.getOrDefault("workDoneProgress", "false").toString());
```

Comment:
The `workDoneProgress` field isn't under `extendedClientCapabilities` though. It's under the regular capabilities.

Reply:
Fixed. I had it this way so I could check and set its value on the client side, but it's probably not necessary. I updated the client code too.

Reply:
Right. Makes sense.